web-common/src/components/column-profile/ColumnProfile.svelte

```diff
@@ -28,33 +28,28 @@
     return () => observer.unobserve(container);
   });
 
-  // get all column profiles.
-  let profileColumns;
-  $: profileColumns = createQueryServiceTableColumns(
-    $runtime?.instanceId,
-    objectName,
-    {},
-    { query: { keepPreviousData: true } }
-  );
+  $: profileColumns =
+    objectName === undefined
+      ? undefined
+      : createQueryServiceTableColumns(
+          $runtime?.instanceId,
+          objectName,
+          {},
+          { query: { keepPreviousData: true } }
+        );
```

Comment:
here and below we just propagate `undefined` if objectName===undefined, because these fns aren't designed to handle undefined input. We then just need to add one more undefined check in the html template and all is good

Reply:
Here we should make use of [TanStack Query's `enabled` flag for dependent queries](https://tanstack.com/query/v4/docs/react/guides/dependent-queries)

```
{ query: { keepPreviousData: true, enabled: !!objectName } }
```

Then we can get rid of these ternaries. Will that work?

Reply:
found another solution -- just pushed the check to upstream components so that <ColumnProfile> is only rendered if `objectName` is a string, in which case this file doesn't need to be changed at all